kornia/color/lab.py

```diff
@@ -0,0 +1,197 @@
+from typing import Tuple
+
+import torch
+import torch.nn as nn
+
+from .xyz import rgb_to_xyz, xyz_to_rgb
+
+"""
+The RGB to Lab color transformations were translated from scikit image's rgb2lab and lab2rgb
+
+https://github.com/scikit-image/scikit-image/blob/a48bf6774718c64dade4548153ae16065b595ca9/skimage/color/colorconv.py
+
+"""
+
+
+def rgb_to_lab(image: torch.Tensor) -> torch.Tensor:
+    r"""Converts a RGB image to Lab.
+
+    The image data is assumed to be in the range of :math:`[0, 1]`. Lab
+    color is computed using the D65 illuminant and Observer 2.
+
+    Args:
+        image (torch.Tensor): RGB Image to be converted to Lab with shape :math:`(*, 3, H, W)`.
+
+    Returns:
+        torch.Tensor: Lab version of the image with shape :math:`(*, 3, H, W)`.
+
+    Example:
+        >>> input = torch.rand(2, 3, 4, 5)
+        >>> output = rgb_to_lab(input)  # 2x3x4x5
+    """
+    if not isinstance(image, torch.Tensor):
+        raise TypeError("Input type is not a torch.Tensor. Got {}".format(
+            type(image)))
+
+    if len(image.shape) < 3 or image.shape[-3] != 3:
+        raise ValueError("Input size must have a shape of (*, 3, H, W). Got {}"
+                         .format(image.shape))
+
+    # Convert from Linear RGB to sRGB
+    r: torch.Tensor = image[..., 0, :, :]
+    g: torch.Tensor = image[..., 1, :, :]
+    b: torch.Tensor = image[..., 2, :, :]
+
+    rs: torch.Tensor = torch.where(r > 0.04045, torch.pow(((r + 0.055) / 1.055), 2.4), r / 12.92)
+    gs: torch.Tensor = torch.where(g > 0.04045, torch.pow(((g + 0.055) / 1.055), 2.4), g / 12.92)
+    bs: torch.Tensor = torch.where(b > 0.04045, torch.pow(((b + 0.055) / 1.055), 2.4), b / 12.92)
+
+    image_s = torch.stack([rs, gs, bs], dim=-3)
+
+    xyz_im: torch.Tensor = rgb_to_xyz(image_s)
+
+    # normalize for D65 white point
+    xyz_ref_white = torch.tensor(
+        [0.95047, 1., 1.08883], device=xyz_im.device, dtype=xyz_im.dtype)[..., :, None, None]
+    xyz_normalized = torch.div(xyz_im, xyz_ref_white)
+
+    power = torch.pow(xyz_normalized, 1 / 3)
+    scale = 7.787 * xyz_normalized + 4. / 29.
+    xyz_int = torch.where(xyz_normalized > 0.008856, power, scale)
+
+    x: torch.Tensor = xyz_int[..., 0, :, :]
+    y: torch.Tensor = xyz_int[..., 1, :, :]
+    z: torch.Tensor = xyz_int[..., 2, :, :]
+
+    L: torch.Tensor = (116. * y) - 16.
+    a: torch.Tensor = 500. * (x - y)
+    b: torch.Tensor = 200. * (y - z)
```

Comment:
mypy complains about this variable since you are overwriting it. To avoid future debugging issues I think it could be renamed to `b_new`. Same for `lab_to_rgb`.